kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriterTest.java

```diff
@@ -0,0 +1,176 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryError;
+import com.google.cloud.bigquery.InsertAllResponse;
+import com.wepay.kafka.connect.bigquery.BigQuerySinkTask;
+import com.wepay.kafka.connect.bigquery.SinkTaskPropertiesFactory;
+import com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig;
+import com.wepay.kafka.connect.bigquery.config.BigQuerySinkTaskConfig;
+import com.wepay.kafka.connect.bigquery.exception.BigQueryConnectException;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.sink.SinkTaskContext;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static com.wepay.kafka.connect.bigquery.BigQuerySinkTaskTest.spoofSinkRecord;
+import static org.mockito.Mockito.*;
+
+public class BigQueryWriterTest {
+    private static SinkTaskPropertiesFactory propertiesFactory;
+
+    @BeforeClass
+    public static void initializePropertiesFactory() {
+        propertiesFactory = new SinkTaskPropertiesFactory();
+    }
+
+    @Test
+    public void testBigQueryNoFailure() {
+        final String topic = "test_topic";
+        final String dataset = "scratch";
+        final Map<String, String> properties = makeProperties("3", "2000", topic, dataset);
+
+        BigQuery bigQuery = mock(BigQuery.class);
+        Map<Long, List<BigQueryError>> emptyMap = mock(Map.class);
+        when(emptyMap.isEmpty()).thenReturn(true);
+
+        InsertAllResponse insertAllResponse = mock(InsertAllResponse.class);
+        when(insertAllResponse.hasErrors()).thenReturn(false);
+        when(insertAllResponse.getInsertErrors()).thenReturn(emptyMap);
+
+        when(bigQuery.insertAll(anyObject()))
+                .thenReturn(insertAllResponse);
+
+        SinkTaskContext sinkTaskContext = mock(SinkTaskContext.class);
+
+        BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, null);
+        testTask.initialize(sinkTaskContext);
+        testTask.start(properties);
+        testTask.put(Collections.singletonList(spoofSinkRecord(topic)));
+        testTask.flush(Collections.emptyMap());
+
+        verify(bigQuery, times(1)).insertAll(anyObject());
+    }
+
+    @Test
+    public void testBigQueryPartialFailure() {
+        final String topic = "test_topic";
+        final String dataset = "scratch";
+        final Map<String, String> properties = makeProperties("3", "2000", topic, dataset);
+
+        Map<Long, List<BigQueryError>> insertErrorMap = mock(Map.class);
+        when(insertErrorMap.isEmpty()).thenReturn(false);
+        when(insertErrorMap.size()).thenReturn(1);
+
+        InsertAllResponse insertAllResponseWithError = mock(InsertAllResponse.class);
```

Comment:
maybe some comments here would be good, like "first attempt (partial failure)" "second attempt (success)"

Reply:
sure